File: arduino/Whether_monitoring_system/Weather.cpp
#include "Weather.h"

Weather::Weather() : Weather(0, 0) {
}
Weather::Weather(int humidity_, int temperature_){
    humidity = humidity_;
    temperature = temperature_;
}
const int Weather::get_humidity() const{
    return this->humidity;
}
void Weather::set_humidity(int humidity_){
    if(humidity_ > 0)
      humidity = humidity_;
}
const int Weather::get_temperature() const{
  return this->temperature;
}
void Weather::set_temperature(int temperature_){
  if(temperature_ > 0)
    temperature = temperature_;
}

File: android/app/src/main/java/org/adeniuobesu/wms/model/Weather.java
package org.adeniuobesu.wms.model;

import java.io.Serializable;

public class Weather implements Serializable {
    private static final long serialVersionUID =1l;

    private int humidity;
    private int temperatureInCelsius;

    public Weather() {
        this(0, 0);
    }
    public Weather(int humidity, int temperatureInCelsius) {
        this.humidity = humidity;
        this.temperatureInCelsius = temperatureInCelsius;
    }

    public int getHumidity() {
        return humidity;
    }

    public int getTemperatureInCelsius() {
        return temperatureInCelsius;
    }

    public void setHumidity(int humidity) {
        this.humidity = humidity;
    }

    public void setTemperatureInCelsius(int celsius) {
        temperatureInCelsius = celsius;
    }

    public int getTemperatureInFahrenheit() {
        return (temperatureInCelsius + 32);
    }

    @Override
    public String toString() {
        return "Weather{humidity=" + humidity
                + ",temperature=" + temperatureInCelsius
                + "}";
    }
}
